sdl2promela/stop_condition/model.py

```diff
@@ -0,0 +1,296 @@
+    """Stop Condition not expression."""
+
+    expression: Expression
+    """Expression after not operator."""
+
+    def __init__(self, expression: Expression):
+        self.expression = expression
+
+
+class PlusExpression(BinaryExpression):
+    """Stop Condition greater plus expression."""
+
+    pass
+
+
+class MinusExpression(BinaryExpression):
+    """Stop Condition greater minus expression."""
+
+    pass
+
+
+class MulExpression(BinaryExpression):
+    """Stop Condition greater multiply  expression."""
+
+    pass
+
+
+class DivExpression(BinaryExpression):
+    """Stop Condition greater divide expression."""
+
+    pass
+
+
+class ModExpression(BinaryExpression):
+    """Stop Condition greater modulo expression."""
+
+    pass
+
+
+class NegationExpression(Expression):
+    """Stop Condition unary negation expression."""
+
+    expression: Expression
+    """Expression after not operator."""
+
+    def __init__(self, expression: Expression):
+        self.expression = expression
+
+
+class IntegerValue(Expression):
+    """Stop Condition integer value expression."""
+
+    value: int
+    """Value of expression."""
+
+    def __init__(self, value: int):
+        self.value = value
+
+
+class FloatValue(Expression):
+    """Stop Condition float value expression."""
+
+    value: float
+    """Value of expression."""
+
+    def __init__(self, value: float):
+        self.value = value
+
+
+class BooleanValue(Expression):
+    """Stop Condition boolean value expression."""
+
+    value: bool
+    """Value of expression."""
+
+    def __init__(self, value: bool):
+        self.value = value
+
+
+class VariableReference(Expression):
+    """Stop Condition reference to single variable."""
+
+    name: str
+    """Name of the variable."""
+
+    def __init__(self, name: str):
+        self.name = name
+
+
+class Selector(Expression):
+    """Stop Condition selector expression.
+
+    Selector expression consists of list of elements separated by '.'.
+    Every element is a simple variable reference
+    or CallExpression, which represents access to the element of array.
+    """
+
+    SelectorElements = List[Union[VariableReference, "CallExpression"]]
+    """Type for list of elements."""
+
+    elements: SelectorElements
+    """List of all elements of selector."""
+
+    def __init__(self, elements: SelectorElements):
+        self.elements = elements
+
+
+class CallExpression(Expression):
+    """Stop Condition call expression.
+
+    Call Expression represents a call to predefined function or,
+    an access to element of array."""
+
+    function: Union[VariableReference, Selector]
+    """Funcion or the selector to the array."""
+    parameters: List[Expression]
+    """List of parameters, or array index."""
+
+    def __init__(self, function: Expression, parameters: List[Expression]):
+        self.function = function
+        self.parameters = parameters
+
+
+class AlwaysStatement:
+    """Stop Condition always statement."""
```

Comment:
Similarly operators -> I assume its easier to have more classes than a type member? I am not saying this is wrong or needs fixing, I am curious about the motivation

Reply:
See answer about operators.